File: October2022/Java/DeleteNodeInALinkedList.java

// @saorav21994

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public void deleteNode(ListNode node) {
        ListNode rem = node.next;
        // System.out.println(rem.val + " " + rem.next.val);
        node.val = rem.val;
        node.next = rem.next;
    }
}

File: October2022/Python/DeleteNodeInALinkedList.py
# https://leetcode.com/problems/delete-node-in-a-linked-list/
# Author: Jiganesh Patil

# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    
    # TC : O(N)
    # SC : O(1)
    def deleteNode(self, node):
        """
        :type node: ListNode
        :rtype: void Do not return anything, modify node in-place instead.
        """
        prev = None
        while node.next:
            node.val = node.next.val
            prev = node
            node = node.next
        prev.next = None
        
    
    # TC : O(1)
    # SC : O(1)
    def deleteNode(self, node):
        """
        :type node: ListNode
        :rtype: void Do not return anything, modify node in-place instead.
        """
        node.val = node.next.val
        node.next = node.next.next